docs/source/guides/data.rst

```diff
@@ -0,0 +1,380 @@
+
+        def training_step(self, batch, batch_idx):
+            # access a dictionnary with a batch from each DataLoader
+            batch_a = batch["a"]
+            batch_b = batch["b"]
+
+
+.. testcode::
+
+    class LitModel(LightningModule):
+
+        def train_dataloader(self):
+
+            loader_a = torch.utils.data.DataLoader(range(8), batch_size=4)
+            loader_b = torch.utils.data.DataLoader(range(16), batch_size=4)
+            loader_c = torch.utils.data.DataLoader(range(32), batch_size=4)
+            loader_c = torch.utils.data.DataLoader(range(64), batch_size=4)
+
+            # pass loaders as a nested dict. This will create batches like this:
+            loaders = {
+                'loaders_a_b': [
+                    loader_a,
+                    loader_b
+                ],
+                'loaders_c_d': {
+                    'c': loader_c,
+                    'd': loader_d
+                }
+            }
+            return loaders
+
+        def training_step(self, batch, batch_idx):
+            # access the data
+            batch_a_b = batch["loaders_a_b"]
+            batch_c_d = batch["loaders_c_d"]
+
+            batch_a = batch_a_b[0]
+            batch_b = batch_a_b[1]
+
+            batch_c = batch_c_d["c"]
+            batch_d = batch_c_d["d"]
+
+----------
+
+Multiple Validation/Test Datasets
+=================================
+For validation and test DataLoaders, you can pass a single DataLoader or a list of them. This optional named
+parameter can be used in conjunction with any of the above use cases. You can choose to pass
+the batches sequentially or simultaneously, as is done for the training step.
+The default mode for validation and test DataLoaders is sequential.
+
+See the following for more details for the default sequential option:
+
+- :meth:`~pytorch_lightning.core.datamodule.LightningDataModule.val_dataloader`
+- :meth:`~pytorch_lightning.core.datamodule.LightningDataModule.test_dataloader`
+
+.. testcode::
+
+    def val_dataloader(self):
+        loader_1 = DataLoader()
+        loader_2 = DataLoader()
+        return [loader_1, loader_2]
+
+To combine batches of multiple test and validation DataLoaders simultaneously, one
+needs to wrap the DataLoaders with `CombinedLoader`.
+
+.. testcode::
+
+    from pytorch_lightning.trainer.supporters import CombinedLoader
+
+    def val_dataloader(self):
+        loader_1 = DataLoader()
+        loader_2 = DataLoader()
+        loaders = {'a': loader_a,'b': loader_b}
+        combined_loaders = CombinedLoader(loaders, "max_size_cycle")
+        return combined_loaders
+
+
+Test with additional data loaders
+=================================
+You can run inference on a test set even if the :func:`~pytorch_lightning.core.Lightning.LightningModule.test_dataloader` method hasn't been
+defined within your :class:`~pytorch_lightning.core.Lightning.LightningModule` instance. For example, rhis would be the case if your test data
```

Comment:
```suggestion
defined within your :class:`~pytorch_lightning.core.Lightning.LightningModule` instance. For example, this would be the case if your test data
```

Reply:
Done !